### PR TITLE
request to publish yolop model

### PR DESCRIPTION
**Pull request by Riser6**
request to publish yolop model， thank you!

### CODE REVIEW

**Comment by NicolasHug** [comment]
@Riser6 thanks for the PR,

I noticed that this is the 3rd PR you opened for YOLOP: please note that this is unnecessary, you can just work on one single PR and edit it as you please by pushing to your development branch (which ideally shouldn't be `master`). Opening many PR is distracting and slows down the overall review process.

I will try to review it some time next week

**Comment by Riser6** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        Sorry for that, I wasn't familiar with your mechanics. And I still work on fixing the hub on CircleCI. Thank you for your advice and apologize for the trouble!Regards!
    
        
            
        
    
    
     
    
        
                    
            
                
                    
                            
                                
                            
                            
                                1528152501
                            
                    
                        
                            
                                    ***@***.***
                            
                        
                
            
        
        
    
    
        签名由
        网易邮箱大师
        定制
    
 

    On 9/3/2021 16:08，Nicolas ***@***.***> wrote： 



@Riser6 thanks for the PR,
I noticed that this is the 3rd PR you opened for YOLOP: please note that this is unnecessary, you can just work on one single PR and edit it as you please by pushing to your development branch (which ideally shouldn't be master). Opening many PR is distracting and slows down the overall review process.
I will try to review it some time next week

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by NicolasHug** [comment]
Just checking in @Riser6 , is this ready for a review? Thanks!

**Comment by Riser6** [comment]
> Just checking in @Riser6 , is this ready for a review? Thanks!

Yes, but my submission failed on CircleCI.  And I tried many times but couldn't fix it. Could you please take a look at it? I'd appreciate it if you could help me!

**Comment by NicolasHug** [comment]
@Riser6 I pushed some minor changes so that our `sanity_checks.py` script passes. Mainly I removed some `:` chars at the end of some section titles, and also replaced a section title with "Model Description" which is a mandatory section (Feel free to put this section somewhere else though!).

Let's see how the CI goes this time.

**Review by NicolasHug — COMMENTED** [review]
Thanks for the PR @Riser6 ,

I took a brief look. According to our [template](https://github.com/pytorch/hub/blob/master/docs/template.md), the submission should contain a small code example illustrating how the model can be used with torchhub:

````
<!-- REQUIRED: provide a working script to demonstrate it works with torch.hub, example below -->
```python
import torch
torch.hub.load('pytorch/vision', 'resnet18', pretrained=True)
```
<!-- Walkthrough a small example of using your model. Ideally, less than 25 lines of code -->
````

Would you mind editing the doc a bit to showcase how to use the model? Feel free to take inspiration from other models like https://pytorch.org/hub/pytorch_vision_fcn_resnet101/

Also, I see that apparently YOLOP relies on old versions of pytorch and torchvision. Ideally, the model should be able to run on the most recent versions. Are there plan to make it work with the latest pytorch / torchvision versions?

**Comment by NicolasHug** [comment]
The model isn't displayed because this PR hasn't beeen merged yet :)

**Comment by Riser6** [comment]
Dear  NicolasHug:
      I just added a code example according to your advice, but  my submission failed on CircleCI again. It seems that I have no solution. Could you please take a look at it? I'd appreciate it if you could help me!

**Comment by NicolasHug** [comment]
Looks like the CI was missing the `prefetch_generator` dependency. I just pushed a commit to add it, let's see if the CI as happy now :)

**Comment by Riser6** [comment]
> Looks like the CI was missing the `prefetch_generator` dependency. I just pushed a commit to add it, let's see if the CI as happy now :)

Thank you very much, but there seem to be some errors that have not been resolved.

**Comment by Riser6** [comment]
> Thanks @Riser6 , I'll look into this
> 
> @brianjo any chance you know how to debug this netlify issue? I can't find any reference to netlify in the repo or on github, so I'm not sure where to start

Thanks a lot! @NicolasHug @brianjo ，Is there a solution now ?

**Comment by Riser6** [comment]
> Thanks @Riser6 , I'll look into this
> 
> @brianjo any chance you know how to debug this netlify issue? I can't find any reference to netlify in the repo or on github, so I'm not sure where to start

Thanks a lot! @NicolasHug ，Is there a solution now ?

**Comment by NicolasHug** [comment]
Perhaps @malfet will know more about this? https://github.com/pytorch/hub/pull/230#issuecomment-953641947

File: hustvl_yolop.md [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+github-link: https://github.com/hustvl/YOLOP
+github-id: hustvl/YOLOP
+featured_image_1: no-image
+featured_image_2: no-image
+accelerator: cuda-optional
+---
+## Before You Start
+Start from a **Python>=3.7** environment with **PyTorch>=1.7** and **torchvision>=0.8** installed. To install YOLOP dependencies:
+```bash
+conda install pytorch==1.7.0 torchvision==0.8.0 cudatoolkit=10.2 -c pytorchconda install pytorch==1.7.0 torchvision==0.8.0 cudatoolkit=10.2 -c pytorch
```
Comment (NicolasHug) [inline hustvl_yolop.md:20]: @Riser6 Do you think these are necessary?

According to the previous test runs, I think the model passes on pytorch latest version - which is preferable, so maybe we can remove these instructions?

Also regarding the previous netlify issue: it's fixed now, we just needed to merge with `master` lol. There's another CI issue that I'm investigating but it's unrelated to this PR :)

The PR is good to go on my side but I'm just wondering if we could remove the instructions above, LMK what you think

File: hustvl_yolop.md [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+```
```
Comment (NicolasHug) [inline hustvl_yolop.md:21]: @Riser6 Do you think these are necessary?

According to the previous test runs, I think the model passes on pytorch latest version - which is preferable, so maybe we can remove these instructions?

Also regarding the previous netlify issue: it's fixed now, we just needed to merge with `master` lol. There's another CI issue that I'm investigating but it's unrelated to this PR :)

The PR is good to go on my side but I'm just wondering if we could remove the instructions above, LMK what you think

**Review by NicolasHug — COMMENTED** [review]
please see below (or above, apparently)

File: hustvl_yolop.md (hunk 2 above)
Comment (Riser6) [reply hustvl_yolop.md:21]: OK, thank you very much, our model was trained with PyTorch =1.7 version, but the latest version should also be ok, because I think pyTorch has good forward compatibility.

Comment (Riser6) [reply hustvl_yolop.md:21]: So I want to know when will this model be posted on your official website? Thanks!

**Comment by Riser6** [comment]
hello @NicolasHug  when will our yolop model be  posted on your official website? Thanks!

File: hustvl_yolop.md (hunk 2 above)
Comment (NicolasHug) [reply hustvl_yolop.md:21]: Before merging would you mind removing these lines then? thanks

Comment (Riser6) [reply hustvl_yolop.md]: I have removed these lines,  looking forward to your merging! Thanks!

**Comment by NicolasHug** [comment]
Merged! Thank you very much for your patience @Riser6 .

The model will be published on the website https://pytorch.org/hub/ at the next pytorch release - that should be some time in Feb.
